security/pfSense-pkg-nmap/files/usr/local/www/nmap_scan.php

```diff
@@ -0,0 +1,468 @@
+<?php
+/*
+ * nmap_scan.php
+ *
+ * part of pfSense (https://www.pfsense.org)
+ * Copyright (c) 2022-2022 Rubicon Communications, LLC (Netgate)
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+require_once("guiconfig.inc");
+/* require_once("pfsense-utils.inc"); */
+require_once("ipsec.inc");
+
+if ($_POST['downloadbtn'] == gettext("Download Results")) {
+	$nocsrf = true;
+}
+
+$fp = "/root/";
+$fn = "nmap.result";
+$fe = "nmap.error"; // stderr
+$max_display_size = 50*1024*1024; // 50MB limit on GUI results display. See https://redmine.pfsense.org/issues/9239
+
+$interfaces = get_configured_interface_with_descr();
+if (ipsec_enabled()) {
+	$interfaces['enc0'] = "IPsec";
+}
+$interfaces['lo0'] = "Localhost";
+
+foreach (array('server' => gettext('OpenVPN Server'), 'client' => gettext('OpenVPN Client')) as $mode => $mode_descr) {
+	if (is_array($config['openvpn']["openvpn-{$mode}"])) {
+		foreach ($config['openvpn']["openvpn-{$mode}"] as $id => $setting) {
+			if (!isset($setting['disable'])) {
+				$interfaces['ovpn' . substr($mode, 0, 1) . $setting['vpnid']] = $mode_descr . ": ".htmlspecialchars($setting['description']);
+			}
+		}
+	}
+}
+
+$interfaces = array_merge($interfaces, interface_ipsec_vti_list_all());
+$interfaces = array_merge(array('' => 'Any'), $interfaces);
+
+$scan_types = array(
+	'syn' => gettext('SYN'),
+	'connect' => gettext('TCP connect()'),
+	'icmp' => gettext('Ping'),
+	'udp' => gettext('UDP'),
+	'arp' => gettext('ARP (directly connected networks only!)')
+);
+
+function nmap_get_running_process(){
+	$processcheck = (trim(shell_exec("/bin/ps axw -O pid= | /usr/bin/grep '/usr/local/bin/nmap' | /usr/bin/grep '{$fn}' | /usr/bin/egrep -v '(pflog|grep)'")));
```

Comment:
Seems like this could be simplified a bit. Three greps chained when it could probably be reduced to one, such as `/usr/bin/grep '/usr/local/bin/[n]map.*{$fn}'`